px-context-browser-item.html

```diff
@@ -50,7 +98,120 @@
       canFavorite: {
         type: Boolean,
         value: false
+      },
+      visible: {
+        type: Boolean,
+        value: true
+      }
+    },
+
+    observers: ['_handleFavoritesItemVisible(visible, styleAsFavorite)'],
+
+    _handleFavoritesItemVisible(visible, styleAsFavorite) {
+      if (visible && styleAsFavorite && !this._breadcrumbsEl) {
+        window.requestAnimationFrame(() => {
+          this._breadcrumbsEl = Polymer.dom(this.root).querySelector('#breadcrumbs');
+        });
+      }
+    },
+
+    _computeBreadcrumbsFontStyles(el) {
+      return this.getFontStyles(el);
+    },
+
+    _computeBreadcrumbsWidth(el) {
+      return el.getBoundingClientRect().width;
+    },
+
+    _computeTruncatedBreadcrumbs(breadcrumbs, fontStyles, width) {
+      if (breadcrumbs && fontStyles && width) {
+        return this.truncateBreadcrumbs(breadcrumbs, fontStyles, width);
+      }
+    },
+
+    /**
+     * Reads the computed font styles (font family name and font size) from the targeted
+     * element and returns an object with the details.
+     *
+     * @param {HTMLElement} el
+     * @returns {Object} styles
+     * @returns {String} styles.size - The font size as a string with px suffix (e.g. '12px')
+     * @returns {String} styles.family - The CSS font family declaration as a string (e.g. 'Helvetica')
+     */
+    getFontStyles(el) {
+        var styles = window.getComputedStyle(el);
+        var size = styles.fontSize;
+        var family = styles.fontFamily;
+        return {
+            size: size,
+            family: family
+        }
+    },
+
+    /**
+     * Iterates over a list of items and determines which will fit into the
+     * available width and which will not.
+     *
+     * If any items do not fit, backtracks and reduces the available width to
+     * ensure an overflowed icon will also fit.
+     *
+     * @param  {Array} items - A flat Array of items to fit
+     * @param {Object} fontStyles - An object with the font size and family strings from CSSText
+     * @param  {Number} width - The width to fit the items into
+     * @return {Array.<Array>} - First entry is Array of items that fit, second entry is Array of items that do not fit
+     */
+    truncateBreadcrumbs(items, fontStyles, width) {
```

Comment:
Should be a private method (e.g. `_truncateBreadcrumbs`) right?

Reply:
:+1: